katdal/sensordata.py

```diff
@@ -249,37 +376,49 @@ def dummy_sensor_data(name, value=None, dtype=np.float64, timestamp=0.0):
             value = ''
         elif np.issubdtype(dtype, np.bool):
             value = False
+        elif np.issubdtype(dtype, np.object):
+            value = None
     else:
         dtype = np.array(value).dtype
-    data = np.rec.fromarrays([[timestamp], [value], ['nominal']],
-                             dtype=[('timestamp', np.float64), ('value', dtype), ('status', '|S7')])
-    return SensorData(data, name)
+    data = np.array(zip([timestamp], [value]),
+                    dtype=[('timestamp', np.float64), ('value', dtype)])
+    return RecordSensorData(data, name)
```

Comment:
If the dtype is object, should the value be wrapped in `ComparableArrayWrapper`, or is this dummy used post-unwrapping?